### PR TITLE
data encoding: tbl (Text Dictionary)

### PR DESCRIPTION
**Pull request by gucio321**
Hi there,
This is related with #909 and https://github.com/OpenDiablo2/HellSpawner/issues/87

### CODE REVIEW

**Comment by essial** [comment]
Conflicts from other merges.

**Comment by gucio321** [comment]
ok, fixed

File: d2common/d2fileformats/d2tbl/text_dictionary.go [hunk 1]
```diff
@@ -156,12 +193,78 @@ func LoadTextDictionary(dictionaryData []byte) (TextDictionary, error) {
...
 
 	return lookupTable, nil
 }
+
+// Marshal encodes text dictionary back to byte slice
+func (td *TextDictionary) Marshal() []byte {
+	// create stream writter
+	sw := d2datautils.CreateStreamWriter()
+
+	sw.PushBytes(td.crcBytes...)
```
Comment (gravestench) [inline d2common/d2fileformats/d2tbl/text_dictionary.go:213]: is the CRC never calculated... ?

Comment (gravestench) [reply d2common/d2fileformats/d2tbl/text_dictionary.go:213]: https://godoc.org/github.com/howeyc/crc16

File: d2common/d2fileformats/d2tbl/text_dictionary.go [hunk 2]
```diff
@@ -89,64 +102,88 @@ func (td TextDictionary) loadHashEntry(idx int, hashEntry *textDictionaryHashEnt
...
+		Entries: make(map[string]string),
+	}
 
 	br := d2datautils.CreateStreamReader(dictionaryData)
 
 	// skip past the CRC
-	_, _ = br.ReadBytes(crcByteCount)
-
-	var err error
+	lookupTable.crcBytes, err = br.ReadBytes(crcByteCount)
```
Comment (gravestench) [inline d2common/d2fileformats/d2tbl/text_dictionary.go:149]: we may want to validate the CRC during decode if we go to the trouble of adding the CRC during encode

File: d2common/d2fileformats/d2tbl/text_dictionary.go (hunk 1 above)
Comment (gucio321) [reply d2common/d2fileformats/d2tbl/text_dictionary.go:213]: I think, it isn't related with this PR.
I've opened a separated issue for it https://github.com/OpenDiablo2/OpenDiablo2/issues/1043

Comment (gravestench) [reply d2common/d2fileformats/d2tbl/text_dictionary.go:213]: would be a nice unit test

**Review by gravestench — APPROVED** [review]

**Pull request merged**